File: app/lib/models/ComplianceReport.dart
/// UC-25/UC-26: a compliance/safety report filed against a driver, and its
/// review/appeal state.
class ComplianceReport {
  final int id;
  final int driverId;
  final String driverName;
  final String category; // traffic | cargo_damage | complaint | criminal | smuggling | forgery | other
  final String description;
  final String? evidenceFilePath;
  final String status; // open | under_review | upheld | dismissed
  final String? resultingAction; // none | warning | suspension | ban
  final String? appealText;
  final String appealStatus; // none | pending | accepted | rejected
  final String createdAt;

  ComplianceReport({
    required this.id,
    required this.driverId,
    required this.driverName,
    required this.category,
    required this.description,
    required this.evidenceFilePath,
    required this.status,
    required this.resultingAction,
    required this.appealText,
    required this.appealStatus,
    required this.createdAt,
  });

  bool get isPendingReview => status == 'open' || status == 'under_review';
  bool get isUpheld => status == 'upheld';
  bool get canAppeal => isUpheld && appealStatus == 'none';
  bool get hasPendingAppeal => appealStatus == 'pending';

  static const categories = [
    'traffic',
    'cargo_damage',
    'complaint',
    'criminal',
    'smuggling',
    'forgery',
    'other',
  ];

  factory ComplianceReport.fromJson(Map<String, dynamic> json) {
    return ComplianceReport(
      id: json['id'] ?? 0,
      driverId: json['driver_id'] ?? 0,
      driverName: json['driver']?['name']?.toString() ?? '',
      category: json['category']?.toString() ?? 'other',
      description: json['description']?.toString() ?? '',
      evidenceFilePath: json['evidence_file_path']?.toString(),
      status: json['status']?.toString() ?? 'open',
      resultingAction: json['resulting_action']?.toString(),
      appealText: json['appeal_text']?.toString(),
      appealStatus: json['appeal_status']?.toString() ?? 'none',
      createdAt: json['created_at']?.toString() ?? '',
    );
  }
}

File: server/app/Models/ComplianceReport.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ComplianceReport extends Model
{
    const CATEGORIES = ['traffic', 'cargo_damage', 'complaint', 'criminal', 'smuggling', 'forgery', 'other'];

    // Categories that trigger an immediate freeze pending Super Admin
    // review, rather than the normal warning -> suspension escalation.
    const IMMEDIATE_FREEZE_CATEGORIES = ['criminal', 'smuggling', 'forgery'];

    protected $fillable = [
        'driver_id',
        'reported_by_user_id',
        'category',
        'description',
        'evidence_file_path',
        'status',
        'resulting_action',
        'appeal_text',
        'appeal_status',
        'resolved_by_user_id',
        'resolved_at',
    ];

    protected $casts = [
        'resolved_at' => 'datetime',
    ];

    public function driver()
    {
        return $this->belongsTo(Driver::class);
    }

    public function reportedBy()
    {
        return $this->belongsTo(User::class, 'reported_by_user_id');
    }

    public function resolvedBy()
    {
        return $this->belongsTo(User::class, 'resolved_by_user_id');
    }

    public function isImmediateFreezeCategory(): bool
    {
        return in_array($this->category, self::IMMEDIATE_FREEZE_CATEGORIES, true);
    }
}
